go/vt/mysqlctl/backup.go

```diff
@@ -274,15 +274,16 @@ func Restore(
 			return mysql.Position{}, err
 		}
 		// Since this is an empty database make sure we start replication at the beginning
-		if err = mysqld.ResetReplication(ctx); err != nil {
+		if err := mysqld.ResetReplication(ctx); err != nil {
```

Comment:
@deepthi Is there any reason we can't just short-circuit and return if either `ResetReplication` or `PopulateMetadataTables` fail? We already do that for `Wait` above. Why is this logic special in the first place?

It seems reasonable at first glance that if these fail, we won't leave mysqld in the state that the caller expects, so we should return an error other than ErrNoBackup (which is actually not an error but a special kind of success).

That's optional for this PR though. I think the form it's in now is the clearest way to express the logic that was already there before this PR. This PR isn't really about that part of the code anyway.

Reply:
That is a good point. I missed this nuance when I reviewed it just now. Since this PR isn't about that part of the code, I think it should _not_ change the behavior of this part of the code.

Reply:
The code previously was wrong. It reported an error when it successfully reset Replication. I found that when testing, it should be fixed

Reply:
it is _supposed_ to return ErrNoBackup if there is no valid backup, even if ResetReplication succeeds.

Reply:
+1

Reply:
Ok I just looked through the history of this and it looks like it has never been correct. In the beginning the code would return ErrNoBackup if PopulateMetadata succeeded or nil if it failed, that seems to be wrong to start. I added in the reset replication to fix a error where MariaDB would not start, that made the logic even more confusing since I set err = ErrNoBackup if that succeeded which changed the return profile, to an also wrong state where it would return ErrNoBackup as long as ResetReplication, or PopulateMetadata succeeded, or it would return the error from ResetReplication if PopulateMetadata failed.

I think this way is now less incorrect then it has ever been, we _always_ return ErrNoBackup, and we inform the user if there is any error with either ResetRepliation or PopulateMetadata

Reply:
Agreed. I came to the same conclusion after going back and looking through the history of these changes.